zk_toolbox/crates/zk_inception/src/commands/prover/init.rs

```diff
@@ -0,0 +1,159 @@
+use common::{check_prover_prequisites, cmd::Cmd, logger, spinner::Spinner};
+use config::EcosystemConfig;
+use xshell::{cmd, Shell};
+use zksync_config::{
+    configs::{object_store::ObjectStoreMode, GeneralConfig},
+    ObjectStoreConfig,
+};
+
+use super::{
+    args::init::{ProofStorageConfig, ProofStorageGCSCreateBucket, ProverInitArgs},
+    utils::get_link_to_prover,
+};
+use crate::messages::{
+    MSG_CHAIN_NOT_FOUND_ERR, MSG_DOWNLOADING_SETUP_KEY_SPINNER, MSG_GENERAL_CONFIG_NOT_FOUND_ERR,
+    MSG_PROOF_COMPRESSOR_CONFIG_NOT_FOUND_ERR, MSG_PROVER_CONFIG_NOT_FOUND_ERR,
+    MSG_PROVER_INITIALIZED,
+};
+
+const PROVER_STORE_MAX_RETRIES: u16 = 10;
+
+pub(crate) async fn run(args: ProverInitArgs, shell: &Shell) -> anyhow::Result<()> {
+    check_prover_prequisites(shell);
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+    let chain_config = ecosystem_config
+        .load_chain(Some(ecosystem_config.default_chain.clone()))
+        .expect(MSG_CHAIN_NOT_FOUND_ERR);
+    let mut general_config = chain_config
+        .get_zksync_general_config()
+        .expect(MSG_GENERAL_CONFIG_NOT_FOUND_ERR);
+
+    let project_ids = get_project_ids(shell)?;
+    let home = shell.var("HOME")?;
+    let setup_key_path = get_setup_key_path(&general_config, &ecosystem_config)?;
+
+    let args = args.fill_values_with_prompt(project_ids, &home, &setup_key_path);
+
+    let object_store_config = match args.proof_store {
+        ProofStorageConfig::FileBacked(config) => ObjectStoreConfig {
+            mode: ObjectStoreMode::FileBacked {
+                file_backed_base_path: config.proof_store_dir,
+            },
+            max_retries: PROVER_STORE_MAX_RETRIES,
+            local_mirror_path: None,
+        },
+        ProofStorageConfig::GCS(config) => ObjectStoreConfig {
+            mode: ObjectStoreMode::GCSWithCredentialFile {
+                bucket_base_url: config.bucket_base_url,
+                gcs_credential_file_path: config.credentials_file,
+            },
+            max_retries: PROVER_STORE_MAX_RETRIES,
+            local_mirror_path: None,
+        },
+        ProofStorageConfig::GCSCreateBucket(config) => create_gcs_bucket(shell, config)?,
+    };
+
+    if args.setup_key_config.download_key {
+        download_setup_key(
+            shell,
+            &general_config,
+            &args.setup_key_config.setup_key_path,
+        )?;
+    }
+
+    let mut prover_config = general_config
+        .prover_config
+        .expect(MSG_PROVER_CONFIG_NOT_FOUND_ERR);
+    prover_config.prover_object_store = Some(object_store_config.clone());
+    general_config.prover_config = Some(prover_config);
```

Comment:
let's ask about public object store as well. 
`shallSaveToPublicBucket` should be true and then we have to ask the same questions as for this object store